one/src/lib.rs

```diff
@@ -172,6 +193,25 @@ struct DBOpts {
     /// Path to storage directory.
     #[arg(short, long, default_value=default_directory().into_os_string(), env = "CERAMIC_ONE_STORE_DIR")]
     store_dir: PathBuf,
+    #[arg(long, env = "CERAMIC_ONE_DB_CACHE_SIZE")]
+    /// Value to use for the sqlite cache_size pragma
+    /// Use the negative version, which represents Kib e.g. 20000 = 20 Mb
+    /// Or the postive version, representing pages
+    /// None means the default is used.
+    db_cache_size: Option<i64>,
+    /// Used for pragma mmap_size
+    /// 10737418240: 10 GB of memory mapped IO
+    /// if this is slightly bigger than your db file it can improve read performance
+    /// Set to 0 to disable. None is the default
+    #[arg(long, env = "CERAMIC_ONE_DB_MMAP_SIZE")]
+    db_mmap_size: Option<u64>,
+    /// The maximum number of read only connections in the pool
+    #[arg(long, default_value = "8", env = "CERAMIC_ONE_DB_MAX_CONNECTIONS")]
+    db_max_connections: u32,
+    /// The sqlite temp_store value to use
+    /// 0 = default, 1 = file, 2 = memory
+    #[arg(long, env = "CERAMIC_ONE_DB_TEMP_STORE")]
+    db_temp_store: Option<SqliteTempStore>,
```

Comment:
Do we need to expose these to the end operator? Do we expect these to change based on DB size or some other usage pattern?

Reply:
Honestly, I'm not sure. I _hope_ we don't expect/need anyone to set them, and we can simply rely on the defaults we've configured, however, they're all documented as things to try for sqlite performance tuning that are very dependent on hardware and database size. This way we can at least make modifications without recompiling the application. Would they make more sense under experimental (tuning) features?

Reply:
>Would they make more sense under experimental (tuning) features?

Yes, lets do this. That way we can iterate on finding good defaults. And hopefully we find good defaults that works for the vast majority of deployments. And only if we cannot find a set of defaults that covers a wide range of deployments do we promote these out of experimental.